aspnetcore/fundamentals/http-requests.md

```diff
@@ -272,6 +272,98 @@ An `IHttpClientBuilder` is returned when adding named or typed clients. The <xre
 
 [!code-csharp[Main](http-requests/samples/2.x/HttpClientFactorySample/Startup.cs?name=snippet12)]
 
+## Use IHttpClientFactory in a console app
+
+In a console app, add the following package references to the project:
+
+* [Microsoft.Extensions.Hosting](https://www.nuget.org/packages/Microsoft.Extensions.Hosting)
+* [Microsoft.Extensions.Http](https://www.nuget.org/packages/Microsoft.Extensions.Http)
+
+In the following example:
+
+* <xref:System.Net.Http.IHttpClientFactory> is registered in the [Generic Host's](xref:fundamentals/host/generic-host) service container.
+* `MyService` creates a client factory instance from the service, which is used to create an `HttpClient`. `HttpClient` is used to retrieve a webpage.
+* `Main` creates a scope to execute the service's `GetPage` method and write the first 500 characters of the webpage content to the console.
+
+```csharp
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+class Program
+{
+    static async Task<int> Main(string[] args)
+    {
+        var builder = new HostBuilder()
+            .ConfigureServices((hostContext, services) =>
+            {
+                services.AddHttpClient();
+                services.AddTransient<IMyService, MyService>();
+            }).UseConsoleLifetime();
+
+        var host = builder.Build();
+
+        using (var serviceScope = host.Services.CreateScope())
+        {
+            var services = serviceScope.ServiceProvider;
+
+            try
+            {
+                var myService = services.GetRequiredService<IMyService>();
+                var pageContent = await myService.GetPage();
+
+                Console.WriteLine(pageContent.Substring(0, 500));
+            }
+            catch (Exception ex)
+            {
+                var logger = services.GetRequiredService<ILogger<Program>>();
+                logger.LogError(ex, "An error occurred.");
+            }
+        }
+
+        await host.RunAsync();
```

Comment:
This prevents the console app from exiting until they Ctrl+C. Is that intentional?

I think you should add a comment here about what it does.

Reply:
Either way it would be fine in this example. I'll add the note.

Reply:
mmmmmmmm 🤔 ... maybe not. Since we're writing to console, let's just kill the line.